espressif/vscode-esp-idf-extension
# rm gcovr dep implement json html report

## Description

Remove `gcovr` use in the extension in favor of implementing our own JSON and HTML report from toolchain gcov executables.

## Type of change

- New feature (non-breaking change which adds functionality)

## Steps to test this pull request

Provide a list of steps to test changes in this PR and required output

1. Follow the [code coverage tutorial](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/tutorial/code_coverage.md) or [code coverage](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/COVERAGE.md) documentation.
3. Observe results. After executing the `ESP-IDF: Add Editor Coverage` and `ESP-IDF: Get HTML Coverage Report for Project` should provide similar results.

## How has this been tested?

Manual testing following code coverage tutorial.

**Test Configuration**:
* ESP-IDF Version: 5.0
* OS (Windows,Linux and macOS): macOS

## Checklist
